documents/model_test.go

```diff
@@ -287,18 +287,20 @@ func TestGetDocumentRootTree(t *testing.T) {
 	tree, err := dm.GetDocumentRootTree()
 
 	// Manually constructing the two node tree:
-	signaturesLengthLeaf := sha256.Sum256(append([]byte("signatures.length0"), make([]byte, 32)...))
-	expectedRootHash := sha256.Sum256(append(signaturesLengthLeaf[:], dm.Document.SigningRoot...))
+	signaturesLengthLeaf := sha256.Sum256(append(append(compactProperties[SignaturesField], []byte{48}...), make([]byte, 32)...))
```

Comment:
Looks like with the new precise proof update the order of hashing leaves has changed.

Reply:
But why did you say the order has changed?

Reply:
It's property, value, salt. Was it different before?

Actually []byte{48} -> seems to be the wrong length value here, that to me looks like an ASCII encoded "SPACE" character and not a 0. Though this will be fixed when changing the way we generate the signatures. So not worth fixing I think.

Reply:
within leaves, not within a leaf. Not sure why but if you look at the next line 291 I had to change the order. 
Regarding the ascii code, int(48) is 0 in ASCII, as the node logic for length fields does: `[]byte("0")`. I could transform that to encode an int in bytes today.

Reply:
Actually this should be addressed once we create the signatures as a tree soon.